src/Components/Web.JS/src/Boot.Server.Common.ts

```diff
@@ -18,42 +18,49 @@ import { sendJSDataStream } from './Platform/Circuits/CircuitStreamingInterop';
 import { fetchAndInvokeInitializers } from './JSInitializers/JSInitializers.Server';
 import { WebRendererId } from './Rendering/WebRendererId';
 import { RootComponentManager } from './Services/RootComponentManager';
+import { detachWebRendererInterop } from './Rendering/WebRendererInteropMethods';
+import { CircuitDotNetCallDispatcher } from './Platform/Circuits/CircuitDotNetCallDispatcher';
 
 let renderingFailed = false;
-let hasStarted = false;
+let started = false;
+let circuitActive = false;
+let startCircuitPromise: Promise<boolean> | undefined;
 let connection: HubConnection;
 let circuit: CircuitDescriptor;
+let dotNetDispatcher: CircuitDotNetCallDispatcher;
 let dispatcher: DotNet.ICallDispatcher;
-let userOptions: Partial<CircuitStartOptions> | undefined;
+let renderQueue: RenderQueue;
+let options: CircuitStartOptions;
+let logger: ConsoleLogger;
+let afterRenderCallback: (() => void) | undefined;
```

Comment:
We're really expanding the amount of module-level state here. Is it at all practical to bundle this together into some object that can be instantiated, as a step towards a future where multiple Blazor instances can coexist?

Reply:
Perhaps your other comment at https://github.com/dotnet/aspnetcore/pull/50170/files#r1300507039 already covers this.

Reply:
Yeah, I decided to bite the bullet and give `Boot.Sever.Common.ts` a rework. It should be quite a bit more manageable now.